### ISSUE TITLE
Exception raised during index rotation

### ISSUE THREAD

**Issue by jframeau** [Graylog2/graylog2-server#1002]
CentOS 6.6
graylog-server v1.0.0 (Oracle Corporation 1.7.0_75 / Linux 2.6.32-504.8.1.el6.x86_64) 
elasticsearch-1.3.9

Logs from Graylog
===============
    2015-02-26T14:56:10.985+01:00 INFO  [IndexRotationThread] Deflector index <graylog2_151> should be rotated, Pointing deflector to new index now!
    2015-02-26T14:56:10.985+01:00 INFO  [Deflector] Cycling deflector to next index now.
    2015-02-26T14:56:11.034+01:00 INFO  [Deflector] Cycling from <graylog2_151> to <graylog2_152>
    2015-02-26T14:56:11.034+01:00 INFO  [Deflector] Creating index target <graylog2_152>...
    2015-02-26T14:56:11.915+01:00 INFO  [Deflector] Done!
    2015-02-26T14:56:11.915+01:00 INFO  [Deflector] Pointing deflector to new target index....
    2015-02-26T14:56:11.947+01:00 INFO  [SystemJobManager] Submitted SystemJob <38eda8a2-bdbf-11e4-9ee3-0050568576b3> [org.graylog2.indexer.SetIndexReadOnlyJob]
    2015-02-26T14:56:11.948+01:00 INFO  [RebuildIndexRangesJob] Re-calculating index ranges.
    2015-02-26T14:56:11.948+01:00 INFO  [SystemJobManager] Submitted SystemJob <38edf6c1-bdbf-11e4-9ee3-0050568576b3> [org.graylog2.indexer.ranges.RebuildIndexRangesJob]
    2015-02-26T14:56:11.949+01:00 INFO  [Deflector] Done!
    2015-02-26T14:56:13.570+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_133] in [1448ms].
    2015-02-26T14:56:13.876+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_149] in [305ms].
    2015-02-26T14:56:14.096+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_137] in [219ms].
    2015-02-26T14:56:14.257+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_136] in [161ms].
    2015-02-26T14:56:14.480+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_135] in [222ms].
    2015-02-26T14:56:14.659+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_134] in [178ms].
    2015-02-26T14:56:14.875+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_145] in [215ms].
    2015-02-26T14:56:15.074+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_146] in [198ms].
    2015-02-26T14:56:15.273+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_147] in [198ms].
    2015-02-26T14:56:15.508+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_139] in [235ms].
    2015-02-26T14:56:15.706+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_148] in [197ms].
    2015-02-26T14:56:15.915+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_138] in [208ms].
    2015-02-26T14:56:16.114+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_141] in [198ms].
    2015-02-26T14:56:16.294+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_142] in [180ms].
    2015-02-26T14:56:16.518+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_143] in [223ms].
    2015-02-26T14:56:16.718+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_144] in [200ms].
    2015-02-26T14:56:17.837+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_151] in [1118ms].
    2015-02-26T14:56:18.169+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_150] in [331ms].
    2015-02-26T14:56:18.188+01:00 INFO  [RebuildIndexRangesJob] Calculated range of [graylog2_140] in [18ms].
    2015-02-26T14:56:18.235+01:00 INFO  [RebuildIndexRangesJob] Done calculating index ranges for 19 indices. Took 6112ms.
    2015-02-26T14:56:18.236+01:00 INFO  [SystemJobManager] SystemJob <38edf6c1-bdbf-11e4-9ee3-0050568576b3> [org.graylog2.indexer.ranges.RebuildIndexRangesJob] finished in 6288ms.
    2015-02-26T14:56:20.890+01:00 WARN  [IndexRotationThread] Deflector is pointing to [graylog2_152], not the newest one: [graylog2_151]. Re-pointing.
    2015-02-26T14:56:21.030+01:00 INFO  [IndexRotationThread] Deflector index <graylog2_151> should be rotated, Pointing deflector to new index now!
    2015-02-26T14:56:21.030+01:00 INFO  [Deflector] Cycling deflector to next index now.
    2015-02-26T14:56:21.084+01:00 INFO  [Deflector] Cycling from <graylog2_151> to <graylog2_152>
    2015-02-26T14:56:21.084+01:00 INFO  [Deflector] Creating index target <graylog2_152>...
    2015-02-26T14:56:21.210+01:00 ERROR [IndexRotationThread] Couldn't point deflector to a new index
    org.elasticsearch.indices.IndexAlreadyExistsException: [graylog2_152] already exists

Logs at the same time from EL:
=========================
```
[2015-02-26 14:56:11,607][INFO ][cluster.metadata         ] [Hamilton Slade] [graylog2_152] creating index, cause [api], shards [4]/[0], mappings []
[2015-02-26 14:56:11,889][INFO ][cluster.metadata         ] [Hamilton Slade] [graylog2_152] create_mapping [message]
[2015-02-26 14:56:31,680][INFO ][monitor.jvm              ] [Hamilton Slade] [gc][young][18830][5192] duration [754ms], collections [1]/[1.2s], total [754ms]/[1.4m], memory [1.9gb]->[1.4gb]/[3.9gb], all_pools {[young] [489.6mb]->[3.5mb]/[532.5mb]}{[survivor] [10.3mb]->[38.8mb]/[66.5mb]}{[old] [1.4gb]->[1.4gb]/[3.3gb]}
[2015-02-26 15:10:08,467][INFO ][monitor.jvm              ] [Hamilton Slade] [gc][young][19645][5357] duration [890ms], collections [1]/[1.3s], total [890ms]/[1.5m], memory [1.9gb]->[1.5gb]/[3.9gb], all_pools {[young] [412.2mb]->[8.8mb]/[532.5mb]}{[survivor] [12.9mb]->[5.3mb]/[66.5mb]}{[old] [1.5gb]->[1.5gb]/[3.3gb]}
[2015-02-26 15:11:18,802][WARN ][monitor.jvm              ] [Hamilton Slade] [gc][young][19714][5373] duration [2s], collections [1]/[2.3s], total [2s]/[1.5m], memory [1.8gb]->[1.5gb]/[3.9gb], all_pools {[young] [292.2mb]->[10.6mb]/[532.5mb]}{[survivor] [7.4mb]->[699.7kb]/[66.5mb]}{[old] [1.5gb]->[1.5gb]/[3.3gb]}
[2015-02-26 15:25:05,046][WARN ][monitor.jvm              ] [Hamilton Slade] [gc][young][20538][5541] duration [1.6s], collections [1]/[2.3s], total [1.6s]/[1.6m], memory [2gb]->[1.6gb]/[3.9gb], all_pools {[young] [499.2mb]->[8.3mb]/[532.5mb]}{[survivor] [19.3mb]->[5.5mb]/[66.5mb]}{[old] [1.5gb]->[1.5gb]/[3.3gb]}
```
Thx for this great piece of code

jfr

**Comment by kroepke** [comment]
possibly related to #779

**Comment by alekseyp** [comment]
We are missing data from time to time, and issue is related to index rotation bug. 
https://www.evernote.com/l/AAoZsJubJ-ZOR5Qo9eZ3XRO7zJb7UnM7F98
Is there anyway we can fix it? or setup and work around right now?

**Comment by joschi** [comment]
We've tried to make the index rotation more robust and to reduce the number of edge-cases during index rotation in the upcoming Graylog 1.2.0 release.

For now, just make sure to set `disable_index_range_calculation = true` (https://github.com/Graylog2/graylog2-server/blob/1.1.5/misc/graylog2.conf#L378-384) in your Graylog configuration file.

@alekseyp Which version of Graylog and which version of Elasticsearch are you using?

**Comment by alekseyp** [comment]
@joschi , gonna try it right now.

We are using Graylog  v1.1.5 and Elasticsearch 1.7.1 with 5 nodes.

```
{
  "cluster_name" : "es-cluster",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 6,
  "number_of_data_nodes" : 5,
  "active_primary_shards" : 465,
  "active_shards" : 930,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0
}
```

**Comment by jhaar** [comment]
Just a "me too". In my case I'm still running a test instance of graylog-1.1.5 (ie haven't scoped the hardware well) and the disk I was running elasticsearch on hit 9x% full and elasticsearch refused to carry on using it. Hours later when I noticed, I cleaned out some tmp files, restarted elasticsearch and then graylog-server reported this "Couldn't point deflector to a new index" error reported here.

Don't know if my issue is related to this one - but it might help :-)
